File: frontend/src/app/core/api/model/slot-machine-result-dto.ts
export interface SlotReelResultDto {
    label: string;
    value: string;
}

export interface SlotMachineResultDto {
    movieId: string;
    title: string;
    directorName: string;
    releaseYear: number;
    posterUrl?: string | null;
    runtimeMinutes?: number | null;
    selectionMechanismMetadata: string;
    reelResults: SlotReelResultDto[];
    isJackpot: boolean;
}

File: backend/Frametric.Application/DTOs/Discovery/SlotMachineResultDto.cs
// Frametric — Cinematic Analytics Platform
// Copyright (C) 2026 Jesús J. Otero Martínez <jesusoteromartinez@outlook.com>
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.

using System.Collections.Generic;

namespace Frametric.Application.DTOs.Discovery;

public record SlotReelResultDto(
    string Label,
    string Value
);

public record SlotMachineResultDto(
    Guid MovieId,
    string Title,
    string DirectorName,
    int ReleaseYear,
    string? PosterUrl,
    int? RuntimeMinutes,
    string SelectionMechanismMetadata,
    IReadOnlyList<SlotReelResultDto> ReelResults,
    bool IsJackpot,
    int MatchCount = 0,
    IReadOnlyList<bool>? MatchedReels = null,
    string? Overview = null
);
